stable/drupal/templates/deployment.yaml

```diff
@@ -25,24 +25,27 @@ spec:
         {{- else }}
           value: "no"
         {{- end }}
-        - name: MARIADB_ROOT_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "drupal.mariadb.fullname" . }}
-              key: mariadb-root-password
         - name: MARIADB_HOST
+        {{- if .Values.mariadb.enabled }}
           value: {{ template "drupal.mariadb.fullname" . }}
+        {{- else }}
+          value: {{ default "" .Values.mariadb.mariadbHost | quote }}
+        {{- end }}
         - name: MARIADB_PORT_NUMBER
           value: "3306"
         - name: DRUPAL_DATABASE_NAME
           value: {{ default "" .Values.mariadb.mariadbDatabase | quote }}
         - name: DRUPAL_DATABASE_USER
           value: {{ default "" .Values.mariadb.mariadbUser | quote }}
         - name: DRUPAL_DATABASE_PASSWORD
+        {{- if .Values.mariadb.enabled }}
```

Comment:
we should move this condition to the secret so we're still using a secret key here